core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncRetryableStage.java

```diff
@@ -97,7 +115,19 @@ public void maybeAttemptExecute(CompletableFuture<Response<OutputT>> future) {
             Duration backoffDelay = retryableStageHelper.getBackoffDelay();
             if (!backoffDelay.isZero()) {
                 retryableStageHelper.logBackingOff(backoffDelay);
-                scheduledExecutor.schedule(() -> attemptExecute(future), backoffDelay.toMillis(), MILLISECONDS);
+            }
+
+            OptionalDouble tokenAcquireTimeSeconds = retryableStageHelper.getSendTokenNonBlocking();
+            if (!tokenAcquireTimeSeconds.isPresent()) {
+                future.completeExceptionally(SdkClientException.create("unable to acquire send token"));
```

Comment:
Can we add more context in the error message to help users mitigate this issue?
Something like
https://github.com/aws/aws-sdk-java-v2/blob/master/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestExecutor.java#L334=L350